browser/brave_browser_main_extra_parts.cc

```diff
@@ -84,5 +90,14 @@ void BraveBrowserMainExtraParts::PreMainMessageLoopRun() {
 #if !BUILDFLAG(IS_ANDROID)
   brave::BraveWindowTracker::CreateInstance(g_browser_process->local_state());
   brave::BraveUptimeTracker::CreateInstance(g_browser_process->local_state());
+
+#if BUILDFLAG(ENABLE_BRAVE_VPN)
+  if (brave_vpn::IsBraveVPNEnabled()) {
+    auto* connection_api = brave_vpn::BraveVPNOSConnectionAPI::GetInstance();
+    connection_api->SetSharedUrlLoaderFactory(
```

Comment:
Wdyt about to do something like:
```
DCHECK(brave_vpn::BraveVPNOSConnectionAPI::CreateInstance(g_browser_process->shared_url_loader_factory(), g_browser_process->local_state()));
```

Reply:
`BraveVPNOSConnectionAPI::CreateInstance()` is introduced and it checks both are valid and set to instance.
It's difficult to check both at `GetInstance()` though.